validation_utils.test.js

```diff
@@ -0,0 +1,113 @@
+const test = require('ava');
+const { check, gen, property } = require('testcheck');
+
+var ValidationUtils = require("./validation_utils.js");
+
+/**
+ * hasFieldsExcept
+ */
+
+test('hasFieldsExcept happy path', t => {
+  t.true(ValidationUtils.hasFieldsExcept({foo: 1}, []));
+  t.true(ValidationUtils.hasFieldsExcept({foo: 1}, ["bar"]));
+});
+
+test('hasFieldsExcept empty obj', t => {
+  t.false(ValidationUtils.hasFieldsExcept({}, []));
+  t.false(ValidationUtils.hasFieldsExcept({}, ["foo"]));
+});
+
+test('hasFieldsExcept single fail', t => {
+  t.false(ValidationUtils.hasFieldsExcept({foo: 1}, ["foo"]));
+});
+
+test('hasFieldsExcept mixed bag', t => {
+  t.true(ValidationUtils.hasFieldsExcept({foo: 1, bar: 2}, ["bar", "baz"]));
+});
+
+/**
+ * isInteger
+ */
+
+test('isInteger === Number.isInteger', t => {
+  const result = check(
+    property(
+      gen.any,
+      e => ValidationUtils.isInteger(e) === Number.isInteger(e)
```

Comment:
n00b: Why does ValidationUtils repeat the ```isInteger``` functionality if `Number` already exposes equivalent functionality?

Reply:
i assumed because `Number` only exposes this functionality in ES6, actually i was thinking of a followup to do something like `ValidationUtils.isInteger = Number.isInteger || function ...`